src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +109,48 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
-        commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
+        commandLineArgs += "-target:${getTargetExec()}"
+        commandLineArgs += "-targetdir:${project.buildDir}"
+
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
+        commandLineArgs += ["\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\""]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def nunitIntermediateFilesPath = new File(getIntermediateNunitResultsPath(), "")
```

Comment:
what is this path used for ? (/ will be used for ?)
Ensuring the folder exists should be (/have been) in the nunit plugin, but if you *must* work it around, you can do so in the convention mapping callback itself (see comment below)

Reply:
Here I am not sure what would be the desired behaviour:
- in case of parallelized runs, Nunit generates in reports/intermediate-results-nunit/ a certain number of files and then merges them
- here in OpenCover, Nunit is ran through Open Cover and we generate a certain number of code coverage report files and then using Report Generator we merge them. The question is, do we need to merge the Nunit files or we will just apply at a build time both the nunit and opencover plugins?

This path is used to know where to save those intermediate nunit files (through OpenCover) but the merge part is not handled yet.

Reply:
I think you should merge the nunit files as well, but it starts being complex/messy to implement here.
(Maybe at some point creating an actuall nunit-opencover task will be simpler)

You can probably do something like `task.doLast { new NUnitTestResultsMerger().merge(files, nunit.testReportPath) }` reusing nunit merger class ?

Reply:
Yes, thought about that too, but it doesn't seem to be able to recognize the NunitTestResultMerger class inside this plugin

Reply:
You probably need to add a compile dependency to nunit plugin (in your build.gradle) for that to work, let me know if you don't suceed.
(unless the merger class isn't public, but I think everything is public by default in groovy)

Reply:
Ok, thank you for your help!